### PR TITLE
style(*): improve readability

### PR DESCRIPTION
**Pull request by NekoAria**
> Negated conditions are more difficult to read than positive ones, so it is best to avoid them where we can. By swapping the `if` and `else` conditions around we can invert the condition and make it positive.

### CODE REVIEW

File: src/i18n/__init__.py [hunk 1]
```diff
@@ -54,12 +54,11 @@ def __getitem__(self, lang_code: Optional[str]) -> "_L10N":
...
-        languages = ALL_LANGUAGES if not only_iso_639 else self.__iso_639_d.keys()
-        res = (
-            tuple(self[lang_code][key] for lang_code in languages if self[lang_code].key_exist(key))
-            if not html_escaped else
-            tuple(self[lang_code].html_escaped(key) for lang_code in languages if self[lang_code].key_exist(key))
-        )
+        languages = self.__iso_639_d.keys() if only_iso_639 else ALL_LANGUAGES
+        res = tuple(self[lang_code].html_escaped(key) for lang_code in languages if
+                    self[lang_code].key_exist(key)) if html_escaped else tuple(
+            self[lang_code][key] for lang_code in languages if self[lang_code].key_exist(key))
```
Comment (Rongronggg9) [inline src/i18n/__init__.py:60]:
```suggestion
        all_l10n = tuple(self[lang_code] for lang_code in languages)
        res = tuple(
            l10n.html_escaped(key) if html_escaped else l10n[key]
            for l10n in all_l10n
            if l10n.key_exist(key)
        )
```

File: src/parsing/medium.py [hunk 2]
```diff
@@ -795,27 +804,31 @@ async def estimate_message_counts(self):
...
-        return sum(1 for medium in self._media if not medium.drop_silently)
+        return sum(not medium.drop_silently for medium in self._media)
 
     def __bool__(self) -> bool:
         return any(not medium.drop_silently for medium in self._media)
 
     @property
     def valid_count(self):
-        return sum(1 for medium in self._media if medium.valid and not medium.drop_silently)
+        return sum(medium.valid and not medium.drop_silently for medium in self._media)
```
Comment (Rongronggg9) [inline src/parsing/medium.py:814]: https://github.com/Rongronggg9/RSS-to-Telegram-Bot/blob/1365e320c25ac25330536ad6515cfcc6f042a2bc/src/parsing/medium.py#L87

`None` cannot be cast to an `int`.

Comment (NekoAria) [reply src/parsing/medium.py:814]:
```Python
return sum(medium.valid is True and not medium.drop_silently for medium in self._media)
```

File: src/env.py [hunk 3]
```diff
@@ -214,16 +209,16 @@ def __list_parser(var: Optional[str]) -> list[str]:
...
 del _img_relay_server
 
 # ----- images.weserv.nl config -----
 _images_weserv_nl = os.environ.get('IMAGES_WESERV_NL') or 'https://images.weserv.nl/'
-IMAGES_WESERV_NL: Final = ('https://' if not _images_weserv_nl.startswith('http') else '') \
-                          + _images_weserv_nl \
-                          + ('' if _images_weserv_nl.endswith('/') else '/')
+IMAGES_WESERV_NL: Final = (
+                                  ('' if _images_weserv_nl.startswith('http') else 'https://') + _images_weserv_nl
+                          ) + ('' if _images_weserv_nl.endswith('/') else '/')
```
Comment (Rongronggg9) [inline src/env.py:221]:
```suggestion
IMAGES_WESERV_NL: Final = (
        ('' if _images_weserv_nl.startswith('http') else 'https://')
        + _images_weserv_nl
        + ('' if _images_weserv_nl.endswith('/') else '/')
)
```

File: src/env.py [hunk 4]
```diff
@@ -214,16 +209,16 @@ def __list_parser(var: Optional[str]) -> list[str]:
 
 # ----- img relay server config -----
 _img_relay_server = os.environ.get('IMG_RELAY_SERVER') or 'https://rsstt-img-relay.rongrong.workers.dev/'
-IMG_RELAY_SERVER: Final = ('https://' if not _img_relay_server.startswith('http') else '') \
-                          + _img_relay_server \
-                          + ('' if _img_relay_server.endswith(('/', '=')) else '/')
+IMG_RELAY_SERVER: Final = (
+                                  ('' if _img_relay_server.startswith('http') else 'https://') + _img_relay_server
+                          ) + ('' if _img_relay_server.endswith(('/', '=')) else '/')
```
Comment (Rongronggg9) [inline src/env.py:214]:
```suggestion
IMG_RELAY_SERVER: Final = (
        ('' if _img_relay_server.startswith('http') else 'https://')
        + _img_relay_server
        + ('' if _img_relay_server.endswith(('/', '=')) else '/')
)
```

File: src/command/inner/customization.py [hunk 5]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-                          data=(f'set={sub_or_user.id},interval|{page}{tail}'
-                                if not is_user
-                                else f'set_default=interval{tail}')),
+            Button.inline(
+                f"{i18n[lang]['monitor_interval']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if interval_d else '')
+                + formatting_time(minutes=interval or db.EffectiveOptions.default_interval),
+                data=f'set_default=interval{tail}'
+                if is_user
+                else f'set={sub_or_user.id},interval|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:114]:
```suggestion
                data=(
                    f'set_default=interval{tail}'
                    if is_user
                    else f'set={sub_or_user.id},interval|{page}{tail}'
                ),
```
Readability.

File: src/command/inner/customization.py [hunk 6]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-                          data=(f'set={sub_or_user.id},notify|{page}{tail}'
-                                if not is_user
-                                else f'set_default=notify{tail}')),
+            Button.inline(
+                f"{i18n[lang]['notification']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if notify_d else '')
+                + i18n[lang]['notification_normal' if notify else 'notification_muted'],
+                data=f'set_default=notify{tail}'
+                if is_user
+                else f'set={sub_or_user.id},notify|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:124]: Same.

File: src/command/inner/customization.py [hunk 7]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
+                f"{i18n[lang]['length_limit']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if length_limit_d else '')
+                + (
+                    str(sub_or_user.length_limit)
+                    if length_limit
+                    else i18n[lang]['length_limit_unlimited']
+                ),
+                data=f'set_default=length_limit{tail}'
+                if is_user
+                else f'set={sub_or_user.id},length_limit|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:148]: Same.

File: src/command/inner/customization.py [hunk 8]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-                          data=(f'set={sub_or_user.id},display_media|{page}{tail}'
-                                if not is_user
-                                else f'set_default=display_media{tail}')),
+            Button.inline(
+                f"{i18n[lang]['display_media']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if display_media_d else '')
+                + i18n[lang][f'display_media_{display_media}'],
+                data=f'set_default=display_media{tail}'
+                if is_user
+                else f'set={sub_or_user.id},display_media|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:158]: Same.

File: src/command/inner/customization.py [hunk 9]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-                          data=(f'set={sub_or_user.id},display_title|{page}{tail}'
-                                if not is_user
-                                else f'set_default=display_title{tail}')),
+            Button.inline(
+                f"{i18n[lang]['display_title']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if display_title_d else '')
+                + i18n[lang][f'display_title_{display_title}'],
+                data=f'set_default=display_title{tail}'
+                if is_user
+                else f'set={sub_or_user.id},display_title|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:168]: Same.

File: src/command/inner/customization.py [hunk 10]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-                          data=(f'set={sub_or_user.id},display_via|{page}{tail}'
-                                if not is_user
-                                else f'set_default=display_via{tail}')),
+            Button.inline(
+                f"{i18n[lang]['display_via']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if display_via_d else '')
+                + i18n[lang][f'display_via_{display_via}'],
+                data=f'set_default=display_via{tail}'
+                if is_user
+                else f'set={sub_or_user.id},display_via|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:178]: Same.

File: src/command/inner/customization.py [hunk 11]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-                          data=(f'set={sub_or_user.id},display_author|{page}{tail}'
-                                if not is_user
-                                else f'set_default=display_author{tail}')),
+            Button.inline(
+                f"{i18n[lang]['display_author']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if display_author_d else '')
+                + i18n[lang][f'display_author_{display_author}'],
+                data=f'set_default=display_author{tail}'
+                if is_user
+                else f'set={sub_or_user.id},display_author|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:188]: Same.

File: src/command/inner/customization.py [hunk 12]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
+                if is_user
+                else f'set={sub_or_user.id},link_preview|{page}{tail}',
+            ),
+            Button.inline(
+                f"{i18n[lang]['style']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if style_d else '')
+                + i18n[lang][f'style_{style}'],
+                data=f'set_default=style{tail}'
+                if is_user
+                else f'set={sub_or_user.id},style|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:206]: Same.

File: src/command/inner/customization.py [hunk 13]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-        (
-            Button.inline(f'{i18n[lang]["cancel"]}', data='cancel'),
+        None
+        if is_user
+        else (
+            Button.switch_inline(
+                f"{i18n[lang]['set_custom_title_button']}",
+                query=f'/set_title {sub_or_user.user_id} {sub_or_user.id} '
+                if tail
+                else f'/set_title {sub_or_user.id} ',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:216]: Same.

File: src/command/inner/customization.py [hunk 14]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
+                same_peer=True,
+            ),
+            Button.switch_inline(
+                f"{i18n[lang]['set_custom_hashtags_button']}",
+                query=f'/set_hashtags {sub_or_user.user_id} {sub_or_user.id} '
+                if tail
+                else f'/set_hashtags {sub_or_user.id} ',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:223]: Same.

File: src/command/inner/customization.py [hunk 15]
```diff
@@ -71,125 +71,162 @@ async def get_customization_buttons(sub_or_user: Union[db.Sub, db.User],
...
-                          data=(f'set={sub_or_user.id},style|{page}{tail}'
-                                if not is_user
-                                else f'set_default=style{tail}')),
+            Button.inline(
+                f"{i18n[lang]['link_preview']}: "
+                + (FALLBACK_TO_USER_DEFAULT_EMOJI if link_preview_d else '')
+                + i18n[lang][f'link_preview_{link_preview}'],
+                data=f'set_default=link_preview{tail}'
+                if is_user
+                else f'set={sub_or_user.id},link_preview|{page}{tail}',
```
Comment (Rongronggg9) [inline src/command/inner/customization.py:198]: Same.

File: src/parsing/medium.py [hunk 16]
```diff
@@ -347,12 +347,18 @@ async def type_fallback(self) -> bool:
...
+                    (
+                            f"Medium {self.original_urls[0]}"
+                            + (f' ({self.info})' if self.info else '')
+                            + f" type fallback to '{self.type_fallback_medium.type}'"
+                            + (
+                                ''
+                                if self.typeFallbackAllowSelfUrls
+                                else f'({self.type_fallback_medium.original_urls[0]})'
+                            )
+                    )
```
Comment (Rongronggg9) [inline src/parsing/medium.py:360]: Redundant brackets.

File: src/parsing/medium.py [hunk 17]
```diff
@@ -347,12 +347,16 @@ async def type_fallback(self) -> bool:
...
-                                else ''))
+                logger.debug(
+                            f"Medium {self.original_urls[0]}"
+                            + (f' ({self.info})' if self.info else '')
+                            + f" type fallback to '{self.type_fallback_medium.type}'"
+                            + (
+                                ''
+                                if self.typeFallbackAllowSelfUrls
+                                else f'({self.type_fallback_medium.original_urls[0]})'
+                            )
```
Comment (Rongronggg9) [inline src/parsing/medium.py:358]:
```suggestion
                    f"Medium {self.original_urls[0]}"
                    + (f' ({self.info})' if self.info else '')
                    + f" type fallback to '{self.type_fallback_medium.type}'"
                    + (
                        ''
                        if self.typeFallbackAllowSelfUrls
                        else f'({self.type_fallback_medium.original_urls[0]})'
                    )
```

Redundant indentation.

**Review by Rongronggg9 — APPROVED** [review]
Looks fine.

**Pull request merged**